api/v1alpha1/hostedcluster_types.go

```diff
@@ -680,6 +680,7 @@ type PlatformSpec struct {
 }
 
 // KubevirtPlatformSpec specifies configuration for kubevirt guest cluster installations
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.generateID) || has(self.generateID)", message="Kubevirt GenerateID is required once set"
```

Comment:
what does it mean "_is required once set_" ?
if GenerateID is required, it must not be empty or missing in the first place.
so there is no meaning to "once set" in my opinion.

and same for the validation message below - "is immutable once set" - if that field can't be empty when the resource is created, the value is set on creation, and the "once set" is superfluousץ

Reply:
here's an explanation, and it's where i took the example from. https://kubernetes.io/blog/2022/09/29/enforce-immutability-using-cel/#immutablility-after-first-modification 

The idea is that I want GenerateID to be allowed to be set exactly once, and never changed after that. Including removing the value entirely.

The line you reference `!has(oldSelf.generateID) || has(self.generateID)` is kind of odd. I believe what it's saying is that it's okay for the value to be set only when the old value was empty.